### PR TITLE
GtkWave : Update gtkwave to 3.3.117

### PR DESCRIPTION
**Pull request by langly**
#### Description

Updated GtkWave from 3.3.115 to 3.3.117.

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 14.0 23A344 arm64
Xcode 15.2 15C500b

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by pmetzger** [comment]
CI failed.

**Comment by langly** [comment]
I saw. Seems like it's falling over on one of the dependencies. I'll check later.

**Comment by langly** [comment]
Ah, yeah. Found some time to look at it now. The problem is not in this Portfile, for the macos-11 build, but in gtk-osx-application-common-gtk3. 

From the logfiles: 
`DEBUG: Using compiler 'Xcode Clang'
DEBUG: Executing proc-pre-org.macports.configure-configure-0
DEBUG: Active variants check for source-type install considers depends_fetch depends_extract depends_lib depends_build depends_run: xz gtk3 pkgconfig
DEBUG: Executing proc-pre-org.macports.configure-configure-1
DEBUG: gtk3 is installed with the following variants: +x11
DEBUG:   required: quartz, forbidden:
DEBUG:   rejected, because required variant quartz is missing
Error:
gtk-osx-application-common-gtk3 is meant to be used only in a GTK quartz
development environment but your version of GTK does not
support quartz.  Please make sure that port gtk3 and all its
dependencies are built with variants +quartz and try again.

Error: Failed to configure gtk-osx-application-common-gtk3: gtk3 +quartz not installed.
DEBUG: Error code: NONE
DEBUG: Backtrace: gtk3 +quartz not installed.`

File: cad/gtkwave/Portfile [hunk 1]
```diff
@@ -19,17 +19,17 @@ long_description \
...
+                size    3506108
 
 depends_build-append \
                 port:pkgconfig
 
 depends_lib-append \
                 port:bzip2 \
                 path:lib/pkgconfig/gtk+-2.0.pc:gtk2 \
-                port:gtk-osx-application-gtk2 \
+                port:gtk-osx-application-gtk3 \
```
Comment (reneeotten) [inline cad/gtkwave/Portfile:32]: this doesn't make sense to me. A few lines above you still depend on "gtk2" so why is this now depending on "*-gtk3"? I would think that should be consistent: either gtk2 or gtk3.

**Review by reneeotten — CHANGES_REQUESTED** [review]

**Comment by pmetzger** [comment]
CI is still failing.

**Comment by pmetzger** [comment]
Well, then we'll need the submitter to still fix any remaining issues above. @reneeotten are you happy at this point?

File: cad/gtkwave/Portfile [hunk 2]
```diff
@@ -40,9 +40,6 @@ configure.args-append \
                 --with-tcl=${prefix}/lib \
                 --with-tk=${prefix}/lib \
                 --disable-mime-update \
-                --disable-silent-rules \
```
Comment (reneeotten) [inline cad/gtkwave/Portfile:43]: this should remain disabled - or is it not a configure argument anymore?

**Review by reneeotten — CHANGES_REQUESTED** [review]
Please do not remove "--disable-silent-rules"

File: cad/gtkwave/Portfile (hunk 2 above)
Comment (langly) [reply cad/gtkwave/Portfile:43]: I added back the disable-silent-rules.

**Pull request merged**